src/decoder/mod.rs

```diff
@@ -836,6 +906,15 @@ impl<R: Read + Seek> Decoder<R> {
         mut buffer: DecodingBuffer,
         chunk_index: u32,
         output_width: usize,
+    ) -> TiffResult<()> {
+        self.read_chunk_to_bytes(buffer.as_bytes_mut(), chunk_index, output_width)
+    }
+
+    fn read_chunk_to_bytes(
+        &mut self,
+        buffer: &mut [u8],
+        chunk_index: u32,
+        output_width: usize,
```

Comment:
I think this function should directly take `output_row_stride` rather than requiring it be a multiple of the pixel size